drivers/input/input_ite_it8xxx2_kbd.c

```diff
@@ -522,68 +506,22 @@ static int kscan_it8xxx2_init(const struct device *dev)
 	return 0;
 }
 
-static int kscan_it8xxx2_configure(const struct device *dev,
-					kscan_callback_t callback)
-{
-	struct kscan_it8xxx2_data *data = dev->data;
-
-	if (!callback) {
-		return -EINVAL;
-	}
-
-	/* Setup callback function */
-	data->callback = callback;
-
-	return 0;
-}
-
-static int kscan_it8xxx2_disable_callback(const struct device *dev)
-{
-	struct kscan_it8xxx2_data *data = dev->data;
-
-	/* Disable keyboard scan loop */
-	atomic_set(&data->enable_scan, 0);
-
-	return 0;
-}
-
-static int kscan_it8xxx2_enable_callback(const struct device *dev)
-{
-	struct kscan_it8xxx2_data *data = dev->data;
-
-	/* Enable keyboard scan loop */
-	atomic_set(&data->enable_scan, 1);
-
-	return 0;
-}
-
-static const struct kscan_driver_api kscan_it8xxx2_driver_api = {
-	.config = kscan_it8xxx2_configure,
-	.disable_callback = kscan_it8xxx2_disable_callback,
-	.enable_callback = kscan_it8xxx2_enable_callback,
-};
-
-static const struct kscan_wuc_map_cfg kscan_wuc_0[IT8XXX2_DT_INST_WUCCTRL_LEN(0)] =
-		IT8XXX2_DT_WUC_ITEMS_LIST(0);
+static const struct input_it8xxx2_kbd_wuc_map_cfg
+	input_it8xxx2_kbd_wuc[IT8XXX2_DT_INST_WUCCTRL_LEN(0)] = IT8XXX2_DT_WUC_ITEMS_LIST(0);
 
 PINCTRL_DT_INST_DEFINE(0);
 
-static const struct kscan_it8xxx2_config kscan_it8xxx2_cfg_0 = {
+static const struct input_it8xxx2_kbd_config input_it8xxx2_kbd_cfg = {
 	.base = (struct kscan_it8xxx2_regs *)DT_INST_REG_ADDR_BY_IDX(0, 0),
 	.irq = DT_INST_IRQN(0),
-	.wuc_map_list = kscan_wuc_0,
+	.wuc_map_list = input_it8xxx2_kbd_wuc,
 	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(0),
 	.kso16_gpios = GPIO_DT_SPEC_INST_GET(0, kso16_gpios),
 	.kso17_gpios = GPIO_DT_SPEC_INST_GET(0, kso17_gpios),
 };
 
-static struct kscan_it8xxx2_data kscan_it8xxx2_kbd_data;
+static struct input_it8xxx2_kbd_data input_it8xxx2_kbd_kbd_data;
 
-DEVICE_DT_INST_DEFINE(0,
-			&kscan_it8xxx2_init,
-			NULL,
-			&kscan_it8xxx2_kbd_data,
-			&kscan_it8xxx2_cfg_0,
-			POST_KERNEL,
-			CONFIG_KSCAN_INIT_PRIORITY,
-			&kscan_it8xxx2_driver_api);
+DEVICE_DT_INST_DEFINE(0, &input_it8xxx2_kbd_init, NULL,
```

Comment:
Non-blocking, add a build assert to verify only one instance.  This can be done in a follow up PR.

Reply:
Handling this in https://github.com/zephyrproject-rtos/zephyr/pull/65161